### PR TITLE
BACK-974: convert Hashflow into state polling dex to stabilize rate request

### PR DESCRIPTION
**Pull request by Verisana**

### CODE REVIEW

File: src/lib/fetcher/fetcher.ts [hunk 1]
```diff
@@ -77,10 +89,12 @@ export default class Fetcher<T> {
...
-      this.logger.info(
-        'Results Data:',
-        JSON.stringify((result as any).data, null, 4).substring(0, 500),
-      );
+      if(!(result instanceof Error)) {
+        this.logger.info(
+          'Results Data:',
+          JSON.stringify((result as any).data, null, 4).substring(0, 500),
+        );
+      }
```
Comment (Verisana) [inline src/lib/fetcher/fetcher.ts:97]: Not sure I understand the reasoning behind this change. Can you elaborate, please?

File: src/dex/hashflow/hashflow.ts [hunk 2]
```diff
@@ -82,13 +91,66 @@ export class Hashflow extends SimpleExchange implements IDex<HashflowData> {
...
     this.hashFlowAuthToken = token;
-    this.api = new HashflowApi('taker', 'paraswap', this.hashFlowAuthToken);
+    this.api = new HashflowApi(
+      'taker',
+      HASHFLOW_API_CLIENT_NAME,
+      this.hashFlowAuthToken,
+    );
+
+    this.pricesCacheKey = `${this.dexHelper.config.data.network}_${this.dexKey}_prices`;
+    this.marketMakersCacheKey = `${this.dexHelper.config.data.network}_${this.dexKey}_mms`;
```
Comment (Verisana) [inline src/dex/hashflow/hashflow.ts:103]: I think better to prefix with CACHE_PREFIX as well. It is global for dex-lib to know which service is serving current cache value

**Review by Verisana — COMMENTED** [review]
Beside from this overall looks good to me :)

File: src/dex/hashflow/hashflow.ts (hunk 2 above)
Comment (aburkut) [reply src/dex/hashflow/hashflow.ts]: @Verisana added

File: src/lib/fetcher/fetcher.ts (hunk 1 above)
Comment (aburkut) [reply src/lib/fetcher/fetcher.ts]: @Verisana I moved fix from @skypper 's PR https://github.com/paraswap/paraswap-dex-lib/pull/353/files#diff-c1f4c6763e1a3fc7055ef43348ac06e8bc71935bf34a82cc616e5d976ec4c8d6R83

Comment (Verisana) [reply src/lib/fetcher/fetcher.ts]: Thanks, got it

**Review by aburkut — APPROVED** [review]

File: src/lib/fetcher/fetcher.ts [hunk 3]
```diff
@@ -52,19 +60,29 @@ export default class Fetcher<T> {
...
           info.authenticate(options);
         }
 
         try {
-          const result = await this.requestWrapper.request({
-            timeout: FETCH_TIMEOUT_MS,
-            ...options,
-          });
+          let result;
+          if (info.requestFunc && isFunction(info.requestFunc)) {
```
Comment (mwamedacen) [inline src/lib/fetcher/fetcher.ts:74]: isFunction is already guaranteed by typing construct

**Review by mwamedacen — APPROVED** [review]

File: src/lib/fetcher/fetcher.ts (hunk 3 above)
Comment (aburkut) [reply src/lib/fetcher/fetcher.ts]: @mwamedacen updated

**Pull request merged**